conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -237,11 +237,11 @@ def test_apple_clang_cppstd_defaults(self):
         self.assertEqual(_make_cppstd_default("apple-clang", "8"), "gnu98")
         self.assertEqual(_make_cppstd_default("apple-clang", "9"), "gnu98")
         self.assertEqual(_make_cppstd_default("apple-clang", "10"), "gnu98")
-        self.assertEqual(_make_cppstd_default("apple-clang", "11"), "gnu98")
-        self.assertEqual(_make_cppstd_default("apple-clang", "12"), "gnu98")
-        self.assertEqual(_make_cppstd_default("apple-clang", "13"), "gnu98")
-        self.assertEqual(_make_cppstd_default("apple-clang", "14"), "gnu98")
-        self.assertEqual(_make_cppstd_default("apple-clang", "15"), "gnu98")
+        self.assertEqual(_make_cppstd_default("apple-clang", "11"), "gnu17")
+        self.assertEqual(_make_cppstd_default("apple-clang", "12"), "gnu17")
+        self.assertEqual(_make_cppstd_default("apple-clang", "13"), "gnu17")
+        self.assertEqual(_make_cppstd_default("apple-clang", "14"), "gnu17")
+        self.assertEqual(_make_cppstd_default("apple-clang", "15"), "gnu17")
```

Comment:
Default apple-clang C++ standard is not C++17 (for apple-clang 15 I don't know, but I'm sure it was still C++98 in apple-clang 14 and previous versions).

Reply:
You are right, let's split the "default cppstd" from "detected default" to address this properly